code/datums/ai/ai.dm

```diff
@@ -1,5 +1,17 @@
 /datum/ai
 	var/name
+	var/mob/living/body		// The parent mob we control.
+	var/waiting	= 	0		// The number of process ticks we've been waiting.
+	var/wait_for = 	1		// How many process ticks to wait before processing.
+	
+/datum/ai/New(var/mob/living/_body)
+	body = _body
 
-/datum/ai/proc/process()
-	return PROCESS_KILL
```

Comment:
Why not keep this proc as
```
/datum/ai/proc/process()
    SHOULD_NOT_OVERRIDE(TRUE)
    if(can_process())
        do_process()
```
Would prevent leaking implementation details to users

Reply:
I don't understand this comment.

Reply:
Further suggestion
```
/datum/ai/proc/process(var/wait) // wait is provided by the mob system to /Life() and life could pass that on to the AI datum
    SHOULD_NOT_OVERRIDE(TRUE)
    if(!(body.stat & FUGUE))  // Noting that this still has the weakness of being unable to toggle the FUGUE-state as is
        return FALSE // No AI control, returning false as no processing was considered

    if(can_process())
        do_process()
    return TRUE // TRUE as processing was considered (but not necessarily executed)
```

Reply:
Callers of the AI system shouldn't have to know that can-process should be called before do-process.

Reply:
process() is a terrible name for a function, given a function already exists with the name Process().

Reply:
Don't use casing as a way to escape already declared methods. Rename it entirely or don't use it at all.